src/lib/array.jl

```diff
@@ -158,6 +158,9 @@ function unzip(tuples)
       map(tuple -> tuple[i], tuples)
```

Comment:
we could replace the outer `map` here with `ntuple`. Not sure if that would have same overall effect.
Not sure we ever actually hit this map function

Reply:
Ok tested that on
`@btime (x->last(map(atan, fill(x,100), fill(2x, 100))))'(0.1)`
nothing I can do makes it perform better than it currently does